FantasyFootball.fs

```diff
@@ -1,112 +1,167 @@
+    player.Sprintspeed
 
-    let createTeam (players:JsonValue[]) (pickedFormation:Formation) =
-        let numberOfGoalkeepers,numberOfDefenders,numberOfMidfielders,numberOfAttackers = match pickedFormation with
-                                                                                          | FourFourTwo -> 1,4,4,2
-                                                                                          | FourThreeThree -> 1,4,3,3
-                                                                                          | ThreeFiveTwo -> 1,3,5,2
-                                                                                          | FourFiveOne -> 1,4,5,1
+let createTeam (pickedFormation: Formation) (players: JsonPlayer list) =
+    let numberOfGoalkeepers, numberOfDefenders, numberOfMidfielders, numberOfAttackers =
+        match pickedFormation with
+        | FourFourTwo -> 1, 4, 4, 2
+        | FourThreeThree -> 1, 4, 3, 3
+        | ThreeFiveTwo -> 1, 3, 5, 2
+        | FourFiveOne -> 1, 4, 5, 1
 
-        let goalKeeper =  getStartingPlayers (getAllGoalkeepers players) findBestGoalkeeper numberOfGoalkeepers |> Array.head
-        let defenders =  getStartingPlayers (getAllDefenders players) findBestDefenders numberOfDefenders
-        let midfielders =  getStartingPlayers (getAllMidfielders players) findBestMidfielders numberOfMidfielders
-        let attackers =  getStartingPlayers (getAllAttackers players) findBestAttackers numberOfAttackers
-        
-        {
-            Goalkeeper = goalKeeper
-            Defenders = defenders
-            Midfielder = midfielders
-            Attacker = attackers
-        }
+    let goalkeeper =
+        getStartingPlayers (getAllGoalkeepers players) findBestGoalkeeper numberOfGoalkeepers
+        |> List.tryHead
+        |> Option.defaultWith (fun _ -> failwith "Goalkeeper was not found")
+    
+    { Goalkeeper = goalkeeper
+      Defenders = getStartingPlayers (getAllDefenders players) findBestDefenders numberOfDefenders
+      Midfielder = getStartingPlayers (getAllMidfielders players) findBestMidfielders numberOfMidfielders
+      Attacker = getStartingPlayers (getAllAttackers players) findBestAttackers numberOfAttackers }
 
-    let findBestTeam formation =
-        let players = getPlayers
-        createTeam players formation |> printTeam
-        
-    let pickTeam (stringFormation:string) =
-        match stringFormation with
-        | "442" -> findBestTeam FourFourTwo
-        | "433" -> findBestTeam FourThreeThree
-        | "352" -> findBestTeam ThreeFiveTwo
-        | "451" -> findBestTeam FourFiveOne
-        | _ -> failwith "Unknown formation, please try again!!"
+let findBestTeam formation =
+    [| 1..150 |]
+    |> Array.map (sprintf "https://www.easports.com/fifa/ultimate-team/api/fut/item?page=%i")
+    |> Array.map Players.AsyncLoad
+    |> Async.Parallel
+    |> Async.RunSynchronously
+    |> Array.collect (fun x -> x.Items)
+    |> Array.toList
```

Comment:
@vasily-kirichenko what's the reason for changing the Array to a List here? I assume some function only available on a List?